### PR TITLE
Feat/11379 collectibles improvements

### PR DESCRIPTION
**Pull request by dlipicar**
Part of [#11379](https://github.com/status-im/status-desktop/issues/11379)

Initial PR moving a most of the collectibles-related logic over to status-go.
There's still a bunch of smaller tasks missing, all listed in the epic.

Commits to be reviewed separately:
- [chore: rename and reorder types](https://github.com/status-im/status-go/pull/3734/commits/86b2a605a58bfd136670e0ac2ad76d5ab1acf017)
- [chore: moved scheduler to utils dir](https://github.com/status-im/status-go/pull/3734/commits/e986bb077b809daaca1ef4ebe33bc4ea65238463)
- [chore: use single opensea client instance](https://github.com/status-im/status-go/pull/3734/commits/f30a7066bb8bd803cb3607da49eded334b359358)
- [chore: make opensea client return common types](https://github.com/status-im/status-go/pull/3734/commits/8976542a9e8d5e28d16e767fd134fd3467756482)
- [feat: implement collectibles service](https://github.com/status-im/status-go/pull/3734/commits/1e8a19fcc66ad83cd19e290ddc7824dfed5935d6)
- [feat: trigger collectibles fetch on new account](https://github.com/status-im/status-go/pull/3734/commits/f9c5ae4281d6cf42535bc77e7db78fbb678d3099)
- [chore: moved collectibles ownership cache to db](https://github.com/status-im/status-go/pull/3734/commits/41e44885736e511f65adf84870236b2c97b3df30)
- [feat: parallel collectibles fetching per account and chain](https://github.com/status-im/status-go/pull/3734/commits/bef66f52d40d545f330a1e53aebd879a881fb717)

### CODE REVIEW

File: protocol/communities/manager.go [hunk 1]
```diff
@@ -2284,7 +2281,7 @@ func (m *Manager) GetOwnedERC721Tokens(walletAddresses []gethcommon.Address, tok
 		}
 
 		for _, owner := range walletAddresses {
-			assets, err := client.FetchAllAssetsByOwnerAndContractAddress(owner, contractAddresses, "", 5)
+			assets, err := client.FetchAllAssetsByOwnerAndContractAddress(chainID, owner, contractAddresses, "", 5)
```
Comment (IvanBelyakoff) [inline protocol/communities/manager.go]: Can we make 5 a constant?

File: services/wallet/connection/status.go
Comment (IvanBelyakoff) [inline services/wallet/connection/status.go:1]: awesome that we have it separate now

File: services/wallet/collectibles/collectibles.go [hunk 2]
```diff
@@ -37,6 +37,8 @@ type Manager struct {
 	opensea                           *opensea.Client
 	nftCache                          map[walletCommon.ChainID]map[string]thirdparty.CollectibleData
 	nftCacheLock                      sync.RWMutex
+	ownershipCache                    map[walletCommon.ChainID]map[common.Address][]thirdparty.CollectibleUniqueID
```
Comment (IvanBelyakoff) [inline services/wallet/collectibles/collectibles.go]: May I ask you if this cache must be persistent across restarts?

File: services/wallet/collectibles/service.go [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+			common.Address(account),
+		},
+		Message: message,
+	})
+}
+
+func (s *Service) Start() {
+	go func() {
+		s.serviceContext, s.cancelFn = context.WithCancel(context.Background())
+		s.timer = time.NewTimer(accountOwnershipUpdateInterval)
```
Comment (IvanBelyakoff) [inline services/wallet/collectibles/service.go]: It feels like you wanted a ticker here, up to you.

File: services/wallet/collectibles/service.go [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+			s.resetTimer(accountOwnershipUpdateInterval)
+
+			if update() {
+				return
+			}
+		}
+	}()
+}
+
+func (s *Service) Stop() {
```
Comment (IvanBelyakoff) [inline services/wallet/collectibles/service.go]: I dont think that goroutine started in `Start` gracefully stops here.

**Review by IvanBelyakoff — APPROVED** [review]
Dario, great job! I really like how you split the PR, it was easy to follow the changes, awesome work!
LGTM, with minor comments. Feel free to address them if you find them valid

File: protocol/communities/manager.go (hunk 1 above)
Comment (dlipicar) [reply protocol/communities/manager.go]: This piece of code went away after rebasing, using a different function now

File: services/wallet/collectibles/collectibles.go (hunk 2 above)
Comment (dlipicar) [reply services/wallet/collectibles/collectibles.go]: Planning to put this in a DB table in the near future: https://github.com/status-im/status-desktop/issues/11592
Though the ownership info is something we want to refresh relatively often (it's analog to the token balances)

File: services/wallet/collectibles/service.go (hunk 3 above)
Comment (dlipicar) [reply services/wallet/collectibles/service.go]: Nice catch! I basically reproduced the implementation of the `history` module, but paying closer attention it seems I could make use of the `async/InfiniteCommand` here... Will make the change.

File: services/utils/scheduler.go [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package activity
+package utils
```
Comment (stefandunca) [inline services/utils/scheduler.go]: I was wondering if async isn't a better package for this.

File: services/wallet/collectibles/service.go [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+		}
+
+		if update() {
+			return
+		}
+
+		for range s.timer.C {
+			s.resetTimer(accountOwnershipUpdateInterval)
+
+			if update() {
```
Comment (stefandunca) [inline services/wallet/collectibles/service.go]: Wondering if possible and what happens if the update goes more than the timeout

File: services/wallet/collectibles/service.go [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+			}
+		}
+	}()
+}
+
+func (s *Service) Stop() {
+	if s.cancelFn != nil {
+		s.cancelFn()
+	}
+}
```
Comment (stefandunca) [inline services/wallet/collectibles/service.go]: Please stop the scheduler also to cleanup running tasks
See here: https://github.com/status-im/status-go/blob/8926990c2de4aaa2b0a22f7c22aaf5705357d106/services/wallet/activity/service.go#L159

**Review by stefandunca — APPROVED** [review]
Great job; I'm looking forward to seeing it in action.

File: services/utils/scheduler.go (hunk 5 above)
Comment (dlipicar) [reply services/utils/scheduler.go]: makes sense!

**Comment by dlipicar** [comment]
@stefandunca @IvanBelyakoff please take a look at the latest commit, I think it deals with all your suggestions
[To be squashed](https://github.com/status-im/status-go/pull/3734/commits/7c89bec2ce37ad3811a8afa48683e54cb1b5f826)

File: services/wallet/collectibles/service.go (hunk 4 above)
Comment (IvanBelyakoff) [reply services/wallet/collectibles/service.go]: This version looks nicer to me

File: services/wallet/collectibles/service.go [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+		} else if err == nil {
+			collectibles := result.([]thirdparty.CollectibleData)
+			res.Collectibles = collectibles
+			res.ErrorCode = ErrorCodeSuccess
+		}
+
+		s.sendResponseEvent(EventGetCollectiblesDataDone, res, err)
+	})
+}
+func (s *Service) Start() {
```
Comment (IvanBelyakoff) [inline services/wallet/collectibles/service.go]:
```suggestion
func (s *Service) Start() {
    if s.group != nil {
        return
    }
```

Comment (IvanBelyakoff) [reply services/wallet/collectibles/service.go]: Wdyt?

**Review by IvanBelyakoff — APPROVED** [review]

File: services/wallet/collectibles/service.go (hunk 8 above)
Comment (dlipicar) [reply services/wallet/collectibles/service.go]: Done

**Review by IvanBelyakoff — APPROVED** [review]
LGTM! Great job!

**Pull request merged**